frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -303,27 +299,58 @@ function PlotlyFigure({
     )
   }
 
-  const { data, layout, frames } = spec
-
   const reset = useCallback((): void => {
-    widgetMgr.setElementState(element.id, SELECTIONS, {})
-    widgetMgr.setElementState(element.id, DATA, {})
     widgetMgr.setStringValue(element, "{}", { fromUi: true }, fragmentId)
-  }, [widgetMgr, element, fragmentId])
+
+    // Reset the selection info within the plotly figure
+    setPlotlyFigure({
```

Comment:
I don't even think we need to `setPlotlyFigure` as far as I can tell.

Reply:
I think we don't need it for all situations. In many situations, plotly already seems to reset the selections as expected. But there are a couple of situations where I think we need this: 
- clearing of form
- resetting after some double clicks which are sometimes not properly handled by plotly nativly (that's why `onDoubleClick` was implemented in the other PR or?)

Reply:
`onDoubleClick` are handled within the graph and when I was prototyping, I was just thinking that we need to reset state when double clicking sometimes. However, I guess that calls `onDeselect` so I guess in this PR, we can actually remove `onDoubleClick`.

Reply:
Hm thinking about it more, this seems actually correct.